src/services/matcher.py

```diff
@@ -62,21 +62,41 @@ def create_distances_df(self,
         return closest_cats
 
     def _get_embeddings(self, entities: List[Entity]):
-        all_embeddings = [c.embeddings for c in entities]
+        all_embeddings = []
+        embeddings_belonging = []
+
+        for entity_id, entity in enumerate(entities):
+            for embedding in entity.embeddings:
+                all_embeddings.append(embedding)
+                embeddings_belonging.append(entity)
+
         all_embeddings = np.float32(np.vstack(all_embeddings))
         all_embeddings = normalize(all_embeddings, axis=1, norm="l2")
-        return all_embeddings
+        return all_embeddings, embeddings_belonging
 
     def filter_by_thr(self, closest: List[ClosestCats], thr: float):
         for cl in closest:
             filtered = [(c, d) for c, d in zip(cl.cats, cl.distances) if d < thr]
             cl.cats = [f for f, _ in filtered]
             cl.distances = [f for _, f in filtered]
             yield cl
+
+    @staticmethod
+    def reduce_closest_cats(closest_cats):
+        """deletes duplicates"""
+        reduced_closest_cats = {}
+        for cat in closest_cats:
+            if id(cat) not in reduced_closest_cats:
+                reduced_closest_cats[id(cat)] = cat
```

Comment:
why not set(closest_cats)?

Reply:
do you agree?
use just set() instead of this method?

Reply:
yes? I ll try to add "[g[0] for _, g in groupby(closest_cats, lambda l: l[0])] "